server/webhook.go

```diff
@@ -18,56 +18,74 @@ import (
 	"github.com/mattermost/mattermost-server/model"
 )
 
-type JIRAWebhookIssue struct {
-	Self   string
-	Key    string
-	Fields struct {
-		Assignee    *jira.User
-		Reporter    *jira.User
-		Summary     string
-		Description string
-		Priority    *struct {
-			Id      string
-			Name    string
-			IconURL string
-		}
-		IssueType struct {
-			Name    string
-			IconURL string
-		}
-		Resolution *struct {
-			Id string
-		}
-		Status struct {
-			Id string
-		}
-		Labels []string
-	}
+const (
+	eventIssueCreated = uint64(1 << iota)
+	eventIssueDeleted
+	eventUnresolvedIssueDeleted
+	eventIssueAssigned
+	eventIssueReopened
+	eventIssueResolved
+	eventIssueUpdatedStatus
+	eventIssueUpdatePriority
+	eventIssueUpdateSummary
+	eventIssueUpdateDescription
+	eventIssueUpdateSprint
+	eventIssueUpdateRank
+	eventIssueUpdatedAttachment
+	eventIssueUpdatedLabels
+	eventCommentCreated
+	eventCommentDeleted
+	eventCommentUpdated
+	eventMax = iota
+)
+
+const maskLegacy = eventIssueCreated |
+	eventIssueReopened |
+	eventIssueResolved |
+	eventUnresolvedIssueDeleted
+
+const maskComments = eventCommentCreated |
+	eventCommentDeleted |
+	eventCommentUpdated
+
+const maskDefault = maskLegacy
+
+// The keys listed here can be used in the Webhook URL to control what events
+// are posted to Mattermost. A matching parameter with a non-empty value must
+// be added to turn on the event display.
+var eventParamMasks = map[string]uint64{
```

Comment:
Propose
```
	"updated_assignee":    eventIssueAssigned,                        // updated assignee
	"updated_priority":       eventIssueUpdatePriority,               // updated priority
	"updated_summary":    eventIssueUpdateSummary,          // updated summary
	"updated_description": eventIssueUpdateDescription,     // updated description
	"updated_sprint":          eventIssueUpdateSprint,               // updated sprint
	"updated_rank":             eventIssueUpdateRank,                // updated rank
	"updated_attachment": eventIssueUpdatedAttachments, // updated attachments
	"updated_labels":          eventIssueUpdatedLabels,           // updated labels
	"updated_all":                 ^(-1 << eventMax),                       // all update events
```

1. because Jira handles issue creation/deletion and comment creation/update/deletion separately, so these don't need to be included.

![image](https://user-images.githubusercontent.com/13119842/58277080-ebefac80-7d66-11e9-815c-eb425a159493.png)

2. add all supported events (doesn't hurt to include?)
3. remove issue updated status, since already included by default

Reply:
@jasonblais 
1. 2/5 to keep the param(s) to explicitly control the visibility (vs what's sent down). Do not include comment events in the default mask. Specifically I am concerned about any existing webhooks that are set up to include all events, we should make sure that their traffic does not change materially.
2. Sure. 0/5 I'd prefer shorter param names.
3. Good catch. Will do.

Reply:
Let's make this more clear. I don't think there is a need to confuse this public interface with cryptic parameter names.

Reply:
Honestly, I'm confused. With ignoring my proposal and only considering what's currently implemented in this PR, if I want to send notifications for
 - issue creation
 - issue updates for assignee, sprint change
 - comment creation

What do I need to configure as an admin?

Reply:
`...&updated_created_comment=1&updated_assignee=1&updated_sprint=1`

you will also get "Reopened", "Resolved", and "DeletedUnresolved" - there's no way to turn them off (short of using JQL in Jira)

Reply:
@levb and what if we simply reduced the scope of the supported webhook events to v1.1, and only added the issue assignment event?

Then the admin wouldn't need to configure any of these extra parameters?

This, assuming the admin can control whether comments creation/update/deletion is handled through the attached screen
![image](https://user-images.githubusercontent.com/13119842/58287626-82c86300-7d7f-11e9-8c56-a5b3dbab0b05.png)

Reply:
Chatted with @levb - the proposal is to support the following events by default, configurable in Jira Settings > System > WebHooks, and not document the additional flags.

- Issue: Created (no change)
- Issue: Updated (ADD issue assigned event)
- Issue: Deleted (no change)
- Comment: Created (new)
- Comment: Updated (new)
- Comment: Updated (new)

![image](https://user-images.githubusercontent.com/13119842/58332343-08442580-7e09-11e9-9ed0-cfc9e344f25d.png)